lib/Wireguard/Wrapper.pm

```diff
@@ -0,0 +1,359 @@
+                }
+                else {
+                    if ($STATE_READ_SECTION == TRUE) {
+                        $STATE_EMPTY_SECTION = FALSE;
+                        my ($attr_name, $attr_value) = split_and_trim($line, "=");
+                        if ($attr_name eq $wg_meta_prefix . "Alias") {
+                            if (exists($alias_map{$attr_value})) {
+                                close($fh);
+                                die("Alias '$attr_value' already exists, aborting");
+                            }
+                            $STATE_READ_ALIAS = TRUE;
+                            $alias = $attr_value;
+                        }
+                        push @section_data_order, $attr_name;
+                        $section_data->{$attr_name} = $attr_value;
+                    }
+                    else {
+                        close($fh);
+                        die("Attribute without a section encountered, aborting");
+                    }
+                }
+            }
+            else {
+                # normal attribute
+                if ($STATE_READ_SECTION == TRUE) {
+                    $STATE_EMPTY_SECTION = FALSE;
+                    my ($attr_name, $attr_value) = split_and_trim($line, '=');
+                    if (_is_identifying($attr_name)) {
+                        $STATE_READ_ID = TRUE;
+                        $identifier = $attr_value;
+                    }
+                    push @section_data_order, $attr_name;
+                    $section_data->{$attr_name} = $attr_value;
+                }
+                else {
+                    close($fh);
+                    die("Attribute without a section encountered, aborting");
+                }
+            }
+        }
+        # store last section
+        if ($STATE_READ_ID == FALSE) {
+            die("Section without identifying information found (Private -or PublicKey field")
+        }
+        else {
+            $parsed_wg_config->{$i_name}->{$identifier} = $section_data;
+            $parsed_wg_config->{$i_name}->{$identifier}->{type} = $section_type;
+            $parsed_wg_config->{$i_name}->{serial} = $serial;
+            $parsed_wg_config->{$i_name}->{section_order} = \@section_order;
+            $parsed_wg_config->{$i_name}->{alias_map} = \%alias_map;
+
+            $parsed_wg_config->{$i_name}->{$identifier}->{order} = \@section_data_order;
+            push @section_order, $identifier;
+            if ($STATE_READ_ALIAS == TRUE) {
+                $alias_map{$alias} = $identifier;
+            }
+        }
+        #print Dumper(\%alias_map);
+        #print Dumper(\@section_order);
+        #print Dumper($parsed_wg_config);
+        close($fh);
+    }
+    return ($parsed_wg_config);
+}
+
+sub _decide_state($line, $comment_prefix, $disabled_prefix) {
+
+    #remove leading and tailing white space
+    $line =~ s/^\s+|\s+$//g;
+    if ($line eq "") {
```

Comment:
I would also use the `for($line){` pattern here

Reply:
Not sure how to apply this here since it is always a single line which is passed as argument

Reply:
Something like this

```perl
for ($line) {
 /^$/ && return IS_EMPTY;
 /^\[/ && return IS_SECTION;
 /^\Q${comment_prefix}/ && return IS_WG_META;
 /^\Q${disabled_prefix}/ && do {
        $line =~ s/^$disabled_prefix//g;
        # lets do a little bit of recursion here ;)
        return _decide_state($line, $comment_prefix, $disabled_prefix);
  };
  /^#/ && return IS_COMMENT;
  return IS_NORMAL;
}
```